_posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md

```diff
@@ -0,0 +1,288 @@
+
+        > protected�IContainerBuilder�CreateContainerBuilderWithCoreServices()
+        >
+        > {
+        > 
+        >     //......
+        >
+        > ����builder.AddDefaultODataServices();
+        >
+        > ����//�Set�Uri�resolver�to�by�default�enabling�unqualified�functions/actions�and�case�insensitive�match.
+        >
+        > ����builder.AddService(
+        >
+        > ��������ServiceLifetime.Singleton,
+        >
+        > ��������typeof(ODataUriResolver),
+        >
+        > ��������sp�=\>�new�UnqualifiedODataUriResolver�{EnableCaseInsensitive�=�true});
+        >
+        > ����return�builder;
+        >
+        > }
+
+    -   WebAPI client (per service) can further inject other dependencies
+    (for example, typically, adding the EDM model) through
+    the'configureAction' argument of the following method from
+    HttpConfigurationExtensions:
+
+        > internal�static�IServiceProvider�CreateODataRootContainer(this�HttpConfiguration�configuration, string�routeName,�Action\<IContainerBuilder\>�configureAction)
+
+###  ODataUriParser configuration with injected ODataUriResolver dependency
+
+When WebApi parses the request Uri, instance of ODataDefaultPathHandler is created with associated service provider container, which is further used to create ODataUriParser with injected dependency of ODataUriResolver.
+
+    public�ODataUriParser(IEdmModel�model,�Uri�relativeUri,�IServiceProvider�container)
+
+Enable Case-Insensitive for Custom Uri function
+-----------------------------------------------
+
+One issue is encountered when trying to bind function call token with
+case-insensitive enabled. The reason is that at the very beginning of
+the function BindAsUriFunction() the name token, when case-insensitive
+is enabled, is coerced to lower case (as shown blow), which is valid for
+build-in function (such as 'startswith' and 'geo.distance', etc), but
+might not be valid for custom uri functions.
+
+> private�QueryNode�BindAsUriFunction(FunctionCallToken�functionCallToken,�List\<QueryNode\>�argumentNodes)
+>
+> {
+>
+> ����if�(functionCallToken.Source�!=�null)
+>
+> ����{
+>
+> ��������//�the�parent�must�be�null�for�a�Uri�function.
+>
+> ��������throw�new�ODataException(ODataErrorStrings.FunctionCallBinder\_UriFunctionMustHaveHaveNullParent(functionCallToken.Name));
+>
+> ����}
+>
+> ����string�functionCallTokenName�=�this.state.Configuration.EnableCaseInsensitiveUriFunctionIdentifier�?�functionCallToken.Name.ToLowerInvariant()�:�functionCallToken.Name;
+
+To implement with the correct behavior for enabled case-insensitive:
+
+-   GetUriFunctionSignatures needs to additionally return signatures associated with function names in a dictionary instance.
+
+-   When resolving best match function based on arguments for the invoked function, MatchSignatureToUriFunction will find the best match. Exception is still thrown in case of ambiguity or no matches found.
+
+Work Items
```

Comment:
Work items should only reference work in public github repo.  We should not reference internal urls.

Reply:
Updated to reference GH links. Thanks.

---
In reply to: [197311712](https://github.com/OData/WebApi/pull/1492#discussion_r197311712) [](ancestors = 197311712)